packages/headless/webpack.config.js

```diff
@@ -0,0 +1,176 @@
+const path = require('path');
+const fs = require('fs');
+const {DefinePlugin} = require('webpack');
+const TerserPlugin = require('terser-webpack-plugin');
+
+const base = {
+  mode: getMode(),
+  entry: {
+    headless: './src/index.ts',
+    recommendation: './src/recommendation.index.ts',
+    'product-recommendation': './src/product-recommendation.index.ts',
+    'product-listing': './src/product-listing.index.ts',
+    'case-assist': './src/case-assist.index.ts',
+  },
+  module: {
+    rules: [
+      {
+        test: /\.tsx?$/,
+        use: 'ts-loader',
+        exclude: /node_modules/,
+      },
+    ],
+  },
+  resolve: {
+    extensions: ['.ts', '.js'],
+  },
+  plugins: [define()],
+};
+
+function getMode() {
+  return process.env.NODE_ENV === 'production' ? 'production' : 'development';
+}
+
+function define() {
+  const version = JSON.parse(fs.readFileSync('package.json', 'utf-8')).version;
+  return new DefinePlugin({
+    'process.env.NODE_ENV': JSON.stringify(getMode()),
+    'process.env.VERSION': JSON.stringify(version),
+  });
+}
+
+function getFileName(data) {
+  const name = data.chunk.name;
+  return name === 'headless' ? name : `${name}/headless`;
+}
+
+function browserAlias() {
+  return {
+    'coveo.analytics': path.resolve(
+      __dirname,
+      './node_modules/coveo.analytics/dist/library.es.js'
+    ),
+    'cross-fetch': path.resolve(__dirname, './fetch-ponyfill.js'),
+    'web-encoding': path.resolve(
+      __dirname,
+      './node_modules/web-encoding/src/lib.js'
+    ),
+  };
+}
+
+const browserBase = {
+  ...base,
+  devtool: 'source-map',
+  optimization: {
+    minimizer: [new TerserPlugin({extractComments: false})],
+  },
+};
+
+function buildBrowserUmd(entry) {
+  const [entryName, entryPath] = entry;
+  const umdGlobalName = getUmdGlobalName(entryName);
+
+  return {
+    ...browserBase,
+    entry: {
+      [entryName]: entryPath,
+    },
+    target: 'web',
+    output: {
+      filename: (data) => `${getFileName(data)}.js`,
+      path: path.resolve(__dirname, 'dist/browser'),
+      library: {
+        type: 'umd',
+        name: umdGlobalName,
+      },
+    },
+    resolve: {
+      ...base.resolve,
+      alias: browserAlias(),
+    },
+  };
+}
+
+function getUmdGlobalName(entryName) {
+  const map = {
+    headless: 'CoveoHeadless',
+    recommendation: 'CoveoHeadlessRecommendation',
+    'case-assist': 'CoveoHeadlessCaseAssist',
+    'product-recommendation': 'CoveoHeadlessProductRecommendation',
+    'product-listing': 'CoveoHeadlessProductListing',
+  };
+
+  const globalName = map[entryName];
+
+  if (globalName) {
+    return globalName;
+  }
+
+  throw new Error(
+    `Please configure a UMD global name for the "${entryName}" entry.`
+  );
+}
+
+const browserUmds = Object.entries(base.entry).map(buildBrowserUmd);
```

Comment:
This line negates most of the speed improvement of webpack.

Why is it needed? To allow specifying different UMD global names for the different subpackages (see line 96).

If we use a single variable, we speed up the build. However:
1. It would be breaking to people who are using sub-packages via CDN (probably a small number).
2. If two sub-package CDN scripts are included, I think this would probably cause a problem, but I need to test to find out which one:
  - Will the second global overwrite the first?
  - If the globals are merged, this would still be a problem because sub-packages might have exports with the same name but different implementation. e.g. `buildFacet` in search vs. product-listing use-case.